server/hotrod/src/main/scala/org/infinispan/server/hotrod/CacheDecodeContext.scala

```diff
@@ -261,18 +260,28 @@ class CacheDecodeContext(server: HotRodServer) extends ServerConstants with Log
     * Otherwise it's just considered number of seconds from
     * now and it's returned in milliseconds unit.
     */
-   protected def toMillis(lifespan: Int): Long = {
-      if (lifespan > SecondsInAMonth) {
-         val unixTimeExpiry = TimeUnit.SECONDS.toMillis(lifespan) - System.currentTimeMillis
+   protected def toMillis(param: ExpirationParam): Long = {
```

Comment:
For older versions I would say keep it as is, but for 8.0 I would say we remove this, what do you guys think?